sycl/plugins/unified_runtime/ur/adapters/cuda/usm.cpp

```diff
@@ -236,3 +300,198 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMReleaseExp(ur_context_handle_t Context,
   UR_ASSERT(!HostPtr, UR_RESULT_ERROR_INVALID_VALUE);
   return UR_RESULT_SUCCESS;
 }
+
+umf_result_t USMMemoryProvider::initialize(ur_context_handle_t Ctx,
+                                           ur_device_handle_t Dev) {
+  Context = Ctx;
+  Device = Dev;
+  // There isn't a way to query this in cuda, and there isn't much info on
+  // cuda's approach to alignment or transfer granularity between host and
+  // device. Within UMF this is only used to influence alignment, and since we
+  // discard that in our alloc implementations it seems we can safely ignore
+  // this as well, for now.
+  MinPageSize = 0;
```

Comment:
in L0 we are allocating a small memory, 1B, then query for the page size using the allocProperties API. maybe something similar could be done here (not for this patch necessarily, but a follow-up).

Reply:
I did try to find something similar to that but no such query exists in cuda from what I could tell. The closest I got was people in a forum talking about doing multiple small allocations and comparing the pointers to try and deduce alignment which sounds pretty sketchy to me